pathways/inspections.py

```diff
@@ -225,66 +291,201 @@ def compute_max_inspectable_stems(num_stems, stems_per_box, within_box_pct):
+                        ret.infested_stems_completion += 1
+                        if not detected:
+                            ret.infested_stems_detection += (
+                                1  # Count infested stems in box if not yet detected
+                            )
+                if ret.infested_stems_detection > 0:
+                    detected = (
+                        True  # Update detected variable if infested stems found in box
+                    )
+        else:  # All other stem selection strategies inspected the same way
+            # Empty lists to hold opened boxes indexes, will be duplicates bc box index
+            # computed per inspected stem
+            boxes_opened_completion = []
+            boxes_opened_detection = []
+            # Loop through stems in sorted index list, sort list so inspection
+            # progresses through indexes in ascending order
+            for stem in indexes_to_inspect.sort():
+                boxes_opened_completion.append(
+                    math.floor(stem / stems_per_box)
+                )  # Compute box index number
+                if not detected:
+                    ret.stems_inspected_detection += 1
+                    boxes_opened_detection.append(
+                        math.floor(stem / stems_per_box)
+                    )  # Compute box index number
+                if shipment.stems[stem]:
+                    ret.infested_stems_completion += (
+                        1  # Count every infested stem in sample
+                    )
+                    if not detected:
+                        ret.infested_stems_detection += (
+                            1  # Should be only 1 infested stem if to detection
+                        )
+                        detected = True
+            # Number of boxes opened is number of unique boxes indexes in boxes opened lists
+            ret.boxes_opened_completion = len(set(boxes_opened_completion))
+            ret.boxes_opened_detection = len(set(boxes_opened_detection))
+    elif unit in ["box", "boxes"]:
+        # Partial box inspections allowed to reduce number of stems inspected if desired
+        within_box_pct = config["inspection"]["within_box_pct"]
+        inspect_per_box = int(math.ceil(within_box_pct * stems_per_box))
+        detected = False
+        ret.boxes_opened_completion = n_units_to_inspect
+        ret.stems_inspected_completion = n_units_to_inspect * inspect_per_box
+        for box in indexes_to_inspect:
             if not detected:
-                ret.stems_inspected_detection += 1
-            if stem:  # Count every infested stem in box, to completion within a box
-                ret.infested_stems_completion += 1
+                ret.boxes_opened_detection += 1
+            for stem in (shipment.boxes[box]).stems[
+                0:inspect_per_box
+            ]:  # In each box, loop through first n stems (n = inspect_per_box)
```

Comment:
Putting the comment on a separate line before the for is probably better, since will result in just two lines, not a one long line which needs to be split then to fit into 88 chars. Same for lines 479 and 486 below and many cases above.